lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -293,6 +293,17 @@ def get(self, request, *args, **kwargs):
 
         return Response(serializer.data)
 
+    def finalize_response(self, request, response, *args, **kwargs):
+        """
+        Return the final response, exposing the 'Date' header for computing relative time to the dates in the data.
+        """
+        response = super().finalize_response(request, response, *args, **kwargs)
+        # Adding this header should be moved somewhere global, not just this endpoint
+        exposedHeaders = response.get('Access-Control-Expose-Headers', '')
+        exposedHeaders += ', Date' if exposedHeaders else 'Date'
+        response['Access-Control-Expose-Headers'] = exposedHeaders
+        return response
```

Comment:
This feels like error-prone code (in the sense of the comma separating in a string etc). Since we do this elsewhere, I would prefer to see this done as a shared mixin rather than duplicated.

Reply:
Comma separating in the string is the standard for how this header is to be written, I'm afraid.  I can add a reference to [the docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#access-control-expose-headers).

Currently we do this in one other place; I want to hoist both up into a middleware so the header is added to all responses (and consolidate client consumption to make use of it on all requests) but doing that is out-of-scope for the current ticket.  I can look at a mixin as an intermediate step.

Reply:
Yeah of course the spec is comma separated, I was talking about the code being futzy, not the spec. A mixin makes sense for reducing duplication, until you make middleware yeah.